engine/src/main/java/io/zeebe/engine/processor/workflow/job/JobEventProcessors.java

```diff
@@ -35,10 +35,10 @@ public static JobErrorThrownProcessor addJobProcessors(
         .onEvent(ValueType.JOB, JobIntent.CREATED, new JobCreatedProcessor(workflowState))
         .onEvent(ValueType.JOB, JobIntent.COMPLETED, new JobCompletedEventProcessor(workflowState))
         .onCommand(ValueType.JOB, JobIntent.CREATE, new CreateProcessor(jobState))
-        .onCommand(ValueType.JOB, JobIntent.COMPLETE, new CompleteProcessor(jobState))
-        .onCommand(ValueType.JOB, JobIntent.FAIL, new FailProcessor(jobState))
+        .onCommand(ValueType.JOB, JobIntent.COMPLETE, new CompleteProcessor<>(jobState))
```

Comment:
Can here the correct type be interfered?

Reply:
Yes it could, but it created a problem in calling `commandControl.accept(JobIntent, JobRecord)` because the JobRecord was bounded by `J extends JobRecord` and the value to place their was retrieved using `JobRecord getJob(long)` in JobState, which was not generified. So in the end it was not possible to use this.